pkg/controller/metadata/metadata_controller.go

```diff
@@ -347,6 +349,45 @@ func (r *ReconcileMetadata) generateMetadataSecretData(instance *verifyv1beta1.M
 	return data, nil
 }
 
+// This function determines if we should regenerate the metadata or not.
+func ShouldRegenerate(secretsObj *corev1.Secret, hashOfRequestSpec string) bool {
+	secretsMap := secretsObj.Data
+
+	// If the rest of the config has changed and the hash has changed, regenerate regardless.
+	if secretsObj.ObjectMeta.Annotations[versionAnnotation] != hashOfRequestSpec {
+		return true
+	}
+
+	// Checking the date from the secrets store to see if we need to regenerate the metadata.
+	byteValidityDays := secretsMap[validityDays]
+	byteValidUntil := secretsMap[validUntil]
+
+	// If theres nothing in the map then regenerate metadata.
+	if byteValidityDays == nil || byteValidUntil == nil {
+		return true
+	}
+
+	intValidityDays, errValidityDays := strconv.Atoi(string(byteValidityDays))
+	validUntilTimeStamp, errValidUntil := time.Parse(time.RFC1123Z, string(byteValidUntil))
+
+	if errValidUntil != nil || errValidityDays != nil {
+		return true
+	}
+
+	// We want to regenerate the metadata if it's at least halfway through its lifetime.
+	metadataHalfLife := 1
+	if intValidityDays > 1 {
```

Comment:
This won't work - if half-life is 1 day then the validity is effectively 2 days. The half-life has to be a half if validity is 1 day. So maybe we should use something that's not `int` for the half-life